### PR TITLE
Add docs to firmware-protobuf

### PR DESCRIPTION
**Pull request by monkbroc**
Story details: https://app.shortcut.com/particle/story/95195

Rather than keep including firmware-protobuf as a git submodule and requiring all consumers to convert the definitions to their own format, this PR adds publishing a public npm module `@particle/firmware-protobuf` with the definitions.

It also adds a readme with some basic instructions.

Next steps would be adding the npm package to `particle-usb` and other apps that need these definitions.

Eventually I would also like to publicly document the behavior of the control request interface to make sure we have a place to reference how these control requests are supposed to work, including Device OS compatibility info. The docs should be mostly auto-generated from JS comments in this repo.

Example auto-generated comment in the `dist/index.js` code:
```
                /**
                 * SIM card types.
                 * 
                 * Note: The values of this enum should match the values defined by the `SimType` enum in the firmware.
                 * @name particle.ctrl.cellular.SimType
                 * @enum {number}
                 * @property {number} INVALID_SIM_TYPE=0 INVALID_SIM_TYPE value
                 * @property {number} INTERNAL=1 INTERNAL value
                 * @property {number} EXTERNAL=2 EXTERNAL value
                 */
                cellular.SimType = (function() {
                    var valuesById = {}, values = Object.create(valuesById);
                    values[valuesById[0] = "INVALID_SIM_TYPE"] = 0;
                    values[valuesById[1] = "INTERNAL"] = 1;
                    values[valuesById[2] = "EXTERNAL"] = 2;
                    return values;
                })();
```

### CODE REVIEW

**Review by suda — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "@particle/firmware-protobuf",
+  "version": "1.0.0",
+  "description": "Particle Device OS Protobuf Definitions",
+  "main": "dist/index.js",
+  "scripts": {
+    "build": "mkdir -p dist && pbjs control/*.proto -t static-module -o dist/index.js --no-delimited --no-convert --no-verify",
```
Comment (sergeuz) [inline package.json:7]: `mkdir -p` won't work on macOS. I suggest adding [mkdirp](https://www.npmjs.com/package/mkdirp) as dev dependency and using it instead.

**Review by sergeuz — APPROVED** [review]
Looks great, thank you for doing this! I have a minor suggestion: can we rename this package and, possibly, the repo to `device-os-proto`? That name is less ambiguous and I can totally see this repo becoming a slightly more than a bunch of precompiled protobuf files in the future.

**Review by joegoggins — APPROVED** [review]
Thanks for tackling this Julien! LTGM.

I also like the idea of changing the name. I think it should be `device-os-protobuf`.

**Comment by monkbroc** [comment]
Thanks for the reviews. I'll change the module name. How about `@particle/device-os-control-requests` since we wil likely add utilities and files with ids to this repo. In other words, I'm expecting that some of the functionality of particle-usb that are needed for USB and BLE will move into this module.

File: package.json (hunk 1 above)
Comment (busticated) [reply package.json]: yep, and also in general prefer `fs-extra` ([link](https://github.com/jprichardson/node-fs-extra)) as it collects `mkdirp` and similar utilities without bloating things

**Comment by sergeuz** [comment]
> How about @particle/device-os-control-requests

Sounds good to me. The reason why I suggested a more generic `device-os-proto` is that we can potentially use this repo for the protocol definitions shared between Device OS and Device Service (there's even an unfinished [feature branch](https://github.com/particle-iot/firmware-protobuf/compare/test/cloud_proto) in this repo that adds those definitions), but, if deemed necessary, those definitions can totally live in a separate protocol repo.

**Review by busticated — APPROVED** [review]
nice 🙌 💪 🙏 👍

**Comment by monkbroc** [comment]
OK. I agree with your logic Sergey. I'll change it to `@particle/device-os-proto` as the most generic.

File: package.json (hunk 1 above)
Comment (sergeuz) [reply package.json]: So it's still an issue? I've verified that `mkdir -p` works in the default shell on my wife's Mac, but I vaguely remember that it wasn't always the case.

Comment (busticated) [reply package.json]: yeah that rings a tiny bell for me as well. that said, `man mkdir` on my mac ("Monterey" `v12.0.1`) shows the `-p` flag docs and this is internal tooling so i think we're good :+1:

**Pull request merged**